### PR TITLE
Update Sniffles from 1.0.12 to 2.4

### PR DESCRIPTION
**Pull request by fubar2**
* [x] This PR does something else (explain below)

Update sniffles from 1.0.12 from 2020, to most recent 2.4 https://www.nature.com/articles/s41587-023-02024-y
Nearly all parameter names have changed over 4 years and dozens of new ones have been introduced.

This PR provides parameters to control the core use case (make a VCF with SV from a bam) at https://github.com/fritzsedlazeck/Sniffles?tab=readme-ov-file. 

Could be sniffles2 as the author now calls it if we need a new tool after all this time? Keeping the tool id still allows the current old version to be run if anyone needs it - and one less new tool in the list.

Anyone know anyone using sniffles for long read SV detection? 
New version has features for SV in populations - such as cancer patients, but would need a new .snf datatype, more tests/parameters and someone who really wants to work on it.

### CODE REVIEW

**Comment by bgruening** [comment]
@wm75 @npinter can you please have a look, or let us know if you prefer a separate v2 tool?

**Comment by fubar2** [comment]
> @wm75 @npinter can you please have a look, or let us know if you prefer a separate v2 tool?

Weekly ping: @wm75 @npinter - advice appreciated - do you prefer a new sniffles2 or keep this as sniffles but update to 2.4? 
One way or another, would really like to get this merged so can test on VGP assemblies please.

**Comment by fubar2** [comment]
@bgruening  @wm75 @npinter: The 4 year old IUC version of sniffles on EU is broken. It emits VCF that tabix refuses to index: 

> Unsorted positions on sequence #1: 163301 followed by 163203

Does anyone really want to keep it as a separate broken tool? 
Updating seems like a good idea to me FWIW. It will still be available, bugs and all. 

```
DEBUG:jbrowse:bgzip -c /data/dnb10/galaxy_db/files/1/e/3/dataset_1e30b767-d1bd-43bd-b3b0-556d9f3d74f2.dat  > /data/jwd05e/main/073/596/73596814/outputs/dataset_9e76ca47-d642-4d54-bf6f-6366236275ca_files/sniffles_on_data_70_4
DEBUG:jbrowse:cd /data/jwd05e/main/073/596/73596814/outputs/dataset_9e76ca47-d642-4d54-bf6f-6366236275ca_files && tabix -f -p vcf /data/jwd05e/main/073/596/73596814/outputs/dataset_9e76ca47-d642-4d54-bf6f-6366236275ca_files/sniffles_on_data_70_4
[E::hts_idx_push] Unsorted positions on sequence #1: 163301 followed by 163203
tbx_index_build3 failed: /data/jwd05e/main/073/596/73596814/outputs/dataset_9e76ca47-d642-4d54-bf6f-6366236275ca_files/sniffles_on_data_70_4
Traceback (most recent call last):
  File "/opt/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/fubar/jbrowse2/c4b77883c0a7/jbrowse2/jbrowse2.py", line 1712, in <module>
    for key in keys:
  File "/opt/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/fubar/jbrowse2/c4b77883c0a7/jbrowse2/jbrowse2.py", line 1328, in process_annotations
    self.add_vcf(dataset_path, outputTrackConfig)
  File "/opt/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/fubar/jbrowse2/c4b77883c0a7/jbrowse2/jbrowse2.py", line 975, in add_vcf
    self.subprocess_check_call(cmd)
  File "/opt/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/fubar/jbrowse2/c4b77883c0a7/jbrowse2/jbrowse2.py", line 440, in subprocess_check_call
    subprocess.check_call(command, cwd=self.get_cwd(cwd))
  File "/usr/local/tools/_conda/envs/mulled-v1-9ce393a2a46dd43ab470c21687c5e351db9cceeeba55143258ad640cf9d99ac9/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['tabix', '-f', '-p', 'vcf', '/data/jwd05e/main/073/596/73596814/outputs/dataset_9e76ca47-d642-4d54-bf6f-6366236275ca_files/sniffles_on_data_70_4']' returned non-zero exit status 1.

```

File: tools/sniffles/sniffles.xml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <tool id="sniffles" name="sniffles" version="@TOOL_VERSION@+galaxy0">
```
Comment (bgruening) [inline tools/sniffles/sniffles.xml:1]:
```suggestion
<tool id="sniffles" name="sniffles" version="@TOOL_VERSION@+galaxy0" profile="23.0">
```

**Review by bgruening — APPROVED** [review]

**Comment by mvdbeek** [comment]
Attention: deployment failure!

https://github.com/galaxyproject/tools-iuc/actions/runs/11020518027

**Comment by bebatut** [comment]
There seem to be issue with this tool. It is invalid on the ToolShed: https://toolshed.g2.bx.psu.edu/repository?repository_id=fd2e07886563dc35&changeset_revision=3f6f028f418f

```
sniffles.xml - This file requires an entry in the tool_data_table_conf.xml file. Upload a file named tool_data_table_conf.xml.sample to the repository that includes the required entry to correct this error.
```

**Pull request merged**